### DISCUSSION TITLE
cannot access zzbfm / class file for com.google.android.gms.internal.zzbfm not found - Ionic 6

### DISCUSSION THREAD

**Discussion by seltix5** [HaylLtd/cordova-plugin-background-geolocation#129] [Q&A] [answered]
Hello,
I'm using Ionic 6 with cordova-background-geolocation-plugin and OneSignal and onesignal-cordova-plugin and it does not compile.

**Error**
```
....android\capacitor-cordova-android-plugins\src\main\java\com\marianhello\bgloc\data\BackgroundActivity.java:20: error: cannot access zzbfm
        confidence = activity.getConfidence();
                             ^
  class file for com.google.android.gms.internal.zzbfm not found
```

**To Reproduce**
```Markdown
( this is using Ionic 6 )

ionic start TEST --type angular --package-id=com.test.app blank
cd TEST

ionic cap add android


npm install cordova-background-geolocation-plugin

// apply this fix required for ionic 6 : https://github.com/HaylLtd/cordova-background-geolocation-plugin/issues/122#issuecomment-1271553737
// in android/app/src/main/res/values/strings.xml, add the following:
//   <string name="plugin_bgloc_account_name">@string/app_name</string>
//   <string name="plugin_bgloc_content_authority">$PACKAGE_NAME</string>
//   <string name="plugin_bgloc_account_type">$PACKAGE_NAME.account</string>


// compile test before installing second plugin
ionic cap sync && cd android && gradlew assembleDebug && cd ..


npm install onesignal-cordova-plugin


// compile test
ionic cap sync && cd android && gradlew assembleDebug && cd ..
// or open on android studio with:
ionic cap sync && ionic capacitor run android -l --external
```

**Screenshots**
![image](https://user-images.githubusercontent.com/11258096/206959313-048d4796-f6c1-4bcb-84c0-388d877a7267.png)

**Desktop (please complete the following information):**
 - Windows 10

Thank you.

**Comment by HarelM** [comment]
Please add gms services in your build gradle like the following.
https://github.com/IsraelHikingMap/Site/blob/d89a3a5fc8f3ce12aa31f6bcd1d4b4499185f66f/IsraelHiking.Web/android/build.gradle#L11

**Reply by seltix5** [reply]
hello,
that line of code was already there, that is not the solution apparently.
thank you

![image](https://user-images.githubusercontent.com/11258096/207054131-d7aad4a8-03df-46ba-83c5-7395b8968a50.png)

**Reply by HarelM** [reply]
I'm using this plugin with capacitor, so I know this works. You'll need to investigate your code in order to understand why this doesn't work...

**Reply by seltix5** [reply]
hello, sorry I udnerstand nothing about capacitor.
well, I did try that but I have no idea how this plugin works and what it needs...
i'm just trying to convert my Ionic 5 to Ionic 6
my steps from zero :

```
ionic start TEST --type angular --package-id=com.test.app blank

ionic cap add android

ionic build; npx cap copy; npx cap sync; npx cap open android // build and open into android studio

```
at this point android studio is able to compile and run the app as expected, I then

`npm install cordova-background-geolocation-plugin
`

at this point there is already a fix required, I randomly found it here https://github.com/HaylLtd/cordova-background-geolocation-plugin/issues/122#issuecomment-1271553737

now it does compile again but I have no way to import the classes to actually use the plugin since I need the "ngx" class thing (whatever that is) since without it I found no other way to load the plugin classes....

![image](https://user-images.githubusercontent.com/11258096/207062944-d7e30909-bb3e-4756-82d6-bc1ace367d93.png)

thus I install the other plugin as instructed in the ionic page ( https://ionicframework.com/docs/native/geolocation ) but I supose then that plugin is not compatible?

I know the reference on that page is for the other plugin ( the one I was using on Ionic 5 ) but it was not working either and it was where I found the comments saying to use your plugin instead with ionic 6  ( https://github.com/mauron85/cordova-plugin-background-geolocation/issues/766 )

thank you and sorry for the posts, sorry I found 0 solutions online about this or guides for ionic 6

**Reply by HarelM** [reply]
There's a difference between ionic and capacitor.
Ionic is the angular framework from UI/HTML perspective.
Capacitor is the underlaying communication.
From what you wrote you probably use both and not use Cordova.
Note that when you install plugins the way you did, bad things will happen, including all kind of collision etc.
I would recommend stating a new project, don't install mauron85 or the geolocation plugin, and don't use the @ionic-native packages are they are not needed (for this specific plugin).
The typing for this package is within this package.
See here for example:
https://github.com/IsraelHikingMap/Site/blob/acc3d6ff9804788dea1984f4d3316a5dc97ea1cb/IsraelHiking.Web/src/application/services/geo-location.service.ts#L2
You can also use https://github.com/capacitor-community/background-geolocation which is a lot simpler to use and configure...
Good luck!

**Reply by seltix5** [reply]
Hi,
My background is creating pages with HTML, JS and PHP and thus Ionic was a great option when I needed to create an app. This is my first app and I never used capacitor before and Ionic 5 did not use capacitor by default.

Not sure what you mean, on Ionic 5 cordova plugins was installed as normal with "cordova plugin add cordova-background-geolocation-plugin", but now with Ionic 6 they say to install it with npm. That is the way the method shown in the docs. How should it be done?

It appears the main problem is an incompatibility with `npm install onesignal-cordova-plugin` but I have no idea what to do now :s

**Reply by seltix5** [reply]
I created this issue on onesignal github too : https://github.com/OneSignal/OneSignal-Cordova-SDK/issues/831
Since the problem appear to be incompatibility between the two I have no idea how or who could help solve this...
Thank you.

**Reply by seltix5** [reply]
I just updated the original post with the exact steps to create the project from zero.

**Comment by seltix5** [comment]
hello! any updates on this?

**Reply by HarelM** [reply]
I'm not sure what you expect to happen here.
This plugin works both for cordova and capacitor (I've tested both).
If there's something you would like to suggest, feel free to open a PR.

**Reply by seltix5** [reply]
I was hopping for someone with more knowledge on this to understand why there is an incompatibility between those two and then it probably would be possible to create some workaround. I have no idea how I could even start to try to debug this incompatibility :s

**Comment by seltix5** [comment, marked as answer]
I finally fixed the problem changing the default GOOGLE_PLAY_SERVICES_VERSION from 11+ to 15+ here :https://github.com/HaylLtd/cordova-background-geolocation-plugin/blob/033ee47d8473f568068a2161bed0350fa571f3c5/plugin.xml#L27.
Thank you
